hack/update-codegen.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -o errexit
+#set -o nounset
+#set -o pipefail
+
+
+SCRIPT_ROOT=$(git rev-parse --show-toplevel)
+
+# Grab code-generator version from go.sum.
+CODEGEN_VERSION=$(grep 'k8s.io/code-generator' go.sum | awk '{print $2}' | head -1)
+CODEGEN_PKG=$(echo `go env GOPATH`"/pkg/mod/k8s.io/code-generator@${CODEGEN_VERSION}")
+
+echo ">> Using ${CODEGEN_PKG}"
+
+# code-generator does work with go.mod but makes assumptions about
+# the project living in `$GOPATH/src`. To work around this and support
+# any location; create a temporary directory, use this as an output
+# base, and copy everything back once generated.
+TEMP_DIR=$(mktemp -d)
+cleanup() {
+    echo ">> Removing ${TEMP_DIR}"
+    rm -rf ${TEMP_DIR}
+}
+trap "cleanup" EXIT SIGINT
+
+echo ">> Temporary output directory ${TEMP_DIR}"
+
+# Ensure we can execute.
+chmod +x ${CODEGEN_PKG}/generate-groups.sh
+
+GOPATH=$(go env GOPATH) ${CODEGEN_PKG}/generate-groups.sh deepcopy,defaulter \
+    _ github.com/weaveworks/eksctl/pkg/apis \
```

Comment:
I used the full path `github.com/weaveworks/eksctl/pkg/apis` instead of relative path like `./pkg/apis`, so that for both cases (inside and outside GOPATH), the generated files will be under $output_base/github.com/weaveworks/eksctl/pkg/apis directory.